server/routes.js

```diff
@@ -17,30 +17,25 @@ function configureRoutes(app) {
   app.post('/login', dashboard.submitLogin)
 
   // new to-be-implemented API routes
-  app.get('/api/clients', api.validateGetClients, googleHelpers.paAuthorize, api.handleGetClients)
-  app.get('/api/clients/:clientId', api.validateGetClient, googleHelpers.paAuthorize, api.handleGetClient)
-  app.get('/api/clients/:clientId/buttons', api.validateGetClientButtons, googleHelpers.paAuthorize, api.handleGetClientButtons)
-  app.get('/api/clients/:clientId/buttons/:buttonId', api.validateGetClientButton, googleHelpers.paAuthorize, api.handleGetClientButton)
-  app.get(
-    '/api/clients/:clientId/buttons/:buttonId/sessions',
-    api.validateGetClientButtonSessions,
-    googleHelpers.paAuthorize,
-    api.handleGetClientButtonSessions,
-  )
-  app.get('/api/clients/:clientId/gateways', api.validateGetClientGateways, googleHelpers.paAuthorize, api.handleGetClientGateways)
-  app.get('/api/clients/:clientId/gateways/:gatewayId', api.validateGetClientGateway, googleHelpers.paAuthorize, api.handleGetClientGateway)
-  app.get('/api/clients/:clientId/vitals', api.validateGetClientVitals, googleHelpers.paAuthorize, api.handleGetClientVitals)
-  app.post('/api/clients', api.validateRegisterClient, googleHelpers.paAuthorize, api.handleRegisterClient)
-  app.post('/api/clients/:clientId/buttons', api.validateRegisterClientButton, googleHelpers.paAuthorize, api.handleRegisterClientButton)
-  app.post('/api/clients/:clientId/gateways', api.validateRegisterClientGateway, googleHelpers.paAuthorize, api.handleRegisterClientGateway)
-  app.put('/api/clients/:clientId', api.validateUpdateClient, googleHelpers.paAuthorize, api.handleUpdateClient)
-  app.put('/api/clients/:clientId/buttons/:buttonId', api.validateUpdateClientButton, googleHelpers.paAuthorize, api.handleUpdateClientButton)
-  app.put('/api/clients/:clientId/gateways/:gatewayId', api.validateUpdateClientGateway, googleHelpers.paAuthorize, api.handleUpdateClientGateway)
+  app.get('/api/clients', api.wrapper, api.handleGetClients)
+  app.get('/api/clients/:clientId', api.validateGetClient, api.wrapper, api.handleGetClient)
```

Comment:
The expressJS functions can take any number of functions as parameters. So you can just add in your new wrapper function to run after `paAuthorize`. Like this: `app.get('/api/clients/:clientId', api.validateGetClient, googleHelpers.paAuthorize, api.wrapper, api.handleGetClient)`

Unless there's something that makes this not work that I don't see. Let me know if there is!

Reply:
My reasoning for putting into the wrapper is twofold:
a) In the future, the means of authentication may be changed to allow for clients to access routes under their client
b) With google.paAuthorize in each route, the linter requires each route to be multiline. (As there are many routes, this makes the routes file veryyy long.)